contrib/mpi4py/CMakeLists.txt

```diff
@@ -4,14 +4,14 @@ set(MPI4PY_DIR mpi4py-3.0.3)
 set(MPI4PY_SOURCES ${MPI4PY_DIR}/src/MPI.c)
 set(DL_SOURCES ${MPI4PY_DIR}/src/dynload.c)
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/${MPI4PY_DIR}/src)
-
 add_library(MPI ${MPI4PY_SOURCES})
-target_link_libraries(MPI ${Python3_LIBRARIES} ${VAMPIRTRACE_LIBRARIES} MPI::MPI_CXX)
+target_link_libraries(MPI PUBLIC Python3::Python MPI::MPI_CXX)
+target_include_directories(MPI PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/${MPI4PY_DIR}/src)
```

Comment:
Do we need `BUILD_INTERFACE` here as well?

Reply:
Good catch!